pkg/state/state.go

```diff
@@ -1527,6 +1581,10 @@ func (s *stateManager) addBlocks() (*proto.Block, error) {
 	if err := s.stor.handleStateHashes(height, ids); err != nil {
 		return nil, wrapErr(ModificationError, err)
 	}
+	// Store total amount and rewards for each of new blocks.
+	if err = s.stor.handleRewardsAndTotalWavesAmount(height, ids, rewards); err != nil {
+		return nil, wrapErr(ModificationError, err)
+	}
```

Comment:
I think this data should be saved in the main cycle above, i.e. in the process of block application.